build/build.go

```diff
@@ -67,7 +75,7 @@ func NewBuildContext(installSuffix string, buildTags []string) *build.Context {
 			"netgo",  // See https://godoc.org/net#hdr-Name_Resolution.
 			"purego", // See https://golang.org/issues/23172.
 		),
-		ReleaseTags: build.Default.ReleaseTags,
+		ReleaseTags: releaseTags,
```

Comment:
I'd like to suggest we simplify and reduce allocations by reusing `build.Default.ReleaseTags`. On Go 1.12+ and newer, it's guaranteed to have exactly the values we want, there is just more of them on 1.13 and newer. So, something like:

```
ReleaseTags: build.Default.ReleaseTags[:12],
```

But I think it may be a good idea to move the 12 to be a constant in `compiler/version_check.go`, so all numbers that need to change when adding support for a newer Go standard library are together in one place. In that case:

```
ReleaseTags: build.Default.ReleaseTags[:compiler.GoMajorVersion],
```

And we should re-add a `// +build go1.12` constraint to `version_check.go`, because attempting to build GopherJS with 1.11 or older is not supported and we wan't want to try to catch all the possible problems that it might cause.

@flimzy Are you okay with making these changes?

Reply:
@dmitshur: I believe I've made all suggested changes in this PR now. Let me know if you'd like to see anything else, and if you'd like me to squash commits.

Reply:
Everything looks good, thank you. No need to squash, we'll do that as part of merging this PR.